TeensyDmx.h

```diff
@@ -135,11 +143,54 @@ class TeensyDmx
         setChannels(startAddress - 1, values, length);
     }
 
+    void sendRDMDiscMute(byte *uid);
+    void sendRDMDiscUnMute(byte *uid);
+    void sendRDMDiscUniqueBranch(byte *lower_uid, byte *upper_uid);
+    void sendRDMGetDeviceInfo(byte *uid);
+    void sendRDMGetIdentifyDevice(byte *uid);
+    void sendRDMSetIdentifyDevice(byte *uid, bool identify_state);
+    void sendRDMGetDmxStartAddress(byte *uid);
+    void sendRDMSetDmxStartAddress(byte *uid, uint16_t dmx_address);
+    void sendRDMGetDmxPersonality(byte *uid);
+    void sendRDMSetDmxPersonality(byte *uid, uint8_t personality);
+    void sendRDMGetDmxPersonalityDescription(byte *uid, uint8_t personality);
+    void sendRDMGetManufacturerLabel(byte *uid);
+    void sendRDMGetDeviceLabel(byte *uid);
+    void sendRDMGetDeviceModelDescription(byte *uid);
+    void sendRDMSetResetDevice(byte *uid, uint8_t reset_mode);
+    void sendRDMGetPanInvert(byte *uid);
+    void sendRDMSetPanInvert(byte *uid, bool invert);
+    void sendRDMGetTiltInvert(byte *uid);
+    void sendRDMSetTiltInvert(byte *uid, bool invert);
+    void sendRDMGetPanTiltSwap(byte *uid);
+    void sendRDMSetPanTiltSwap(byte *uid, bool swap);
+    void sendRDMGetFactoryDefaults(byte *uid);
+    void sendRDMSetFactoryDefaults(byte *uid);
+    void sendRDMGetLampState(byte *uid);
+    void sendRDMSetLampState(byte *uid, uint8_t lamp_state);
+    void sendRDMGetLampOnMode(byte *uid);
+    void sendRDMSetLampOnMode(byte *uid, uint8_t mode);
+    void sendRDMGetPowerOnSelfTest(byte *uid);
+    void sendRDMSetPowerOnSelfTest(byte *uid, bool power_on_self_test);
+    void sendRDMGetPerformSelftest(byte *uid);
+    void sendRDMSetPerformSelftest(byte *uid, uint8_t test_number);
+    void sendRDMGetSelfTestDescription(byte *uid, uint8_t test_number);
+    void sendRDMGetSensorDefinition(byte *uid, uint8_t sensor_number);
+    void sendRDMGetSensorValue(byte *uid, uint8_t sensor_number);
+    void sendRDMSetSensorValue(byte *uid, uint8_t sensor_number);
+    void sendRDMSetRecordSensors(byte *uid, uint8_t sensor_number);
+
   private:
     TeensyDmx(const TeensyDmx&);
     TeensyDmx& operator=(const TeensyDmx&);
 
-    enum State { IDLE, BREAK, DMX_TX, DMX_RECV, DMX_COMPLETE, RDM_RECV, RDM_RECV_CHECKSUM_HI, RDM_RECV_CHECKSUM_LO, RDM_RECV_POST_CHECKSUM };
+    enum State { IDLE, BREAK,
```

Comment:
Now there are so many states, it might be worth having some comments on them to describe what they are actually for.  It would certainly make debugging any issues in the future easier.

Reply:
Done